File: 振动模式反馈/FeedbackDemo/ViewController.m
//
//  ViewController.m
//  FeedbackDemo
//
//  Created by Quinn on 2018/10/18.
//  Copyright © 2018 Quinn. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)action:(UIButton *)sender {
    
    //这个方法通过传入参数来确定触发什么样的用户触觉反馈
        switch (sender.tag) {
            case 1:
            {
                UIImpactFeedbackGenerator * imp = [[UIImpactFeedbackGenerator alloc]initWithStyle:UIImpactFeedbackStyleHeavy];
                [imp impactOccurred];
            }
                break;
            case 2:
            {
                UIImpactFeedbackGenerator * imp = [[UIImpactFeedbackGenerator alloc]initWithStyle:UIImpactFeedbackStyleMedium];
                [imp impactOccurred];
            }
                break;
            case 3:
            {
                UIImpactFeedbackGenerator * imp = [[UIImpactFeedbackGenerator alloc]initWithStyle:UIImpactFeedbackStyleLight];
                [imp impactOccurred];
            }
                break;
            case 4:
            {
                UINotificationFeedbackGenerator * imp = [[UINotificationFeedbackGenerator alloc]init];
                [imp notificationOccurred:UINotificationFeedbackTypeError];
            }
                break;
            case 5:
            {
                UINotificationFeedbackGenerator * imp = [[UINotificationFeedbackGenerator alloc]init];
                [imp notificationOccurred:UINotificationFeedbackTypeSuccess];
            }
                break;
            case 6:
            {
                UINotificationFeedbackGenerator * imp = [[UINotificationFeedbackGenerator alloc]init];
                [imp notificationOccurred:UINotificationFeedbackTypeWarning];
            }
                break;
            case 7:
            {
                UISelectionFeedbackGenerator * imp = [[UISelectionFeedbackGenerator alloc]init];
                [imp selectionChanged];
            }
                break;
            default:
                break;
        }
}



@end

File: PhotoCollection(相簿)/PhotoCollection(相簿)/ViewController.swift
//
//  ViewController.swift
//  PhotoCollection(相簿)
//
//  Created by Quinn on 2018/10/8.
//  Copyright © 2018 Quinn. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }


}





func Screen_2_1()->Bool{
    let width = UIScreen.main.bounds.width
    let height = UIScreen.main.bounds.height
    return CGFloat(width/height) <= 0.5
}


let SCREEN_WIDTH = UIScreen.main.bounds.size.width
let SCREEN_HEIGHT = UIScreen.main.bounds.size.height

let WIDTH_SCALE:CGFloat = {
    if  Screen_2_1(){
        return UIScreen.main.bounds.size.width/375
        
    }else{
        return UIScreen.main.bounds.size.width/375
        
    }
}()

let HEIGHT_SCALE :CGFloat = {
    if  Screen_2_1(){
        return UIScreen.main.bounds.size.height/667
    }else{
        return UIScreen.main.bounds.size.height/812
    }
}()
